src/v/config/property.h

```diff
@@ -777,6 +778,45 @@ class one_or_many_map_property
     }
 };
 
+/*
+ * Same as property<std::unordered_set<T>> but will also decode a
+ * single T. This can be useful for dealing with backwards compatibility or
+ * creating easier yaml schemas that have simplified special cases.
+ */
+template<typename T>
+class one_or_many_set_property : public property<std::unordered_set<T>> {
```

Comment:
`one_or_many` are kind of annoying for external tools because they have to understand the two ways of expressing the same value for the length=1 case.

We have a few of these for legacy reasons but for new stuff it's preferable to have a single way of expressing the property.

Reply:
Gotcha, was thinking that the `one_or_many` properties were preferred. Your explanation makes a lot of sense though. Will remove the `one_or_many_set_property` implementation and switch the property to `property<std::unordered_set<ss::sstring>>`.